indigo_app/views/tasks.py

```diff
@@ -327,13 +336,25 @@ def post(self, request, *args, **kwargs):
 
         task.save()
 
+        if self.change == 'close' or self.change == 'cancel':
+            self.update_blocked_tasks(task, user)
+
         return redirect(self.get_redirect_url())
 
     def get_redirect_url(self):
         if self.request.GET.get('next'):
             return self.request.GET.get('next')
         return reverse('task_detail', kwargs={'place': self.kwargs['place'], 'pk': self.kwargs['pk']})
 
+    def update_blocked_tasks(self, task, user):
```

Comment:
This is a good candidate to move onto the task object, since it doesn't really have anything to do with the view. It could either be done based on the signals that django-fsm fires when transitions happen, or the `cancel` and `close` methods on the task could call it explicitly (which is probably the best).

Reply:
This improves the abstraction: the view is responsible for translating the user's browser request into some action: eg. blocking or unblocking, etc. The task object itself is responsible for managing the impact of that action, not the view. The view should focus on the request, validating permissions, validating input and parameters, and then delegate to the object (or something else) to do the actual business logic. It makes testing and code re-use much easier.

Reply:
I was blocked by only having access to the view, not the user, but since refactoring that code this is now possible. I did a little hoop jumping to get the messages to also say which tasks have been updated and unblocked by closing a task, le tmeknow what you think